classes/ActionScheduler_ListTable.php

```diff
@@ -0,0 +1,354 @@
+		self::$should_include_js = true;
+
+		$this->store = (object) array(
+			'action' => ActionScheduler_Store::instance(),
+			'log'    => ActionScheduler_Logger::instance(),
+		);
+
+		parent::__construct( array(
+			'singular' => $this->translate( 'action-scheduler' ),
+			'plural'   => $this->translate( 'action-scheduler' ),
+			'ajax'     => false,
+		) );
+	}
+
+	/**
+	 * Setups the libraries. It is executed if `is_admin()` is TRUE.
+	 */
+	public static function init() {
+		add_action( 'admin_footer', __CLASS__ . '::print_script' );
+		add_action( 'admin_enqueue_scripts', __CLASS__ . '::register_javascript' );
+	}
+
+	/**
+	 * Prints our javascript, if we need it.
+	 */
+	public static function print_script() {
+		if ( ! self::$should_include_js ) {
+			return;
+		}
+		wp_enqueue_script( 'action-scheduler' );
+		wp_print_styles( 'wp-jquery-ui-dialog' );
+	}
+
+	/**
+	 * Registers our javascript library and its dependencies.
+	 */
+	public static function register_javascript() {
+		wp_register_script( 'action-scheduler', plugins_url( 'action-scheduler.js', dirname( __FILE__ ) ), array(
+			'jquery',
+			'jquery-ui-core',
+			'jquery-ui-dialog',
+		), '1.0', true );
+	}
+
+	/**
+	 * Returns the recurrence of an action or 'Non-repeating'. The output is human readable.
+	 *
+	 * @param ActionScheduler_Action $item
+	 *
+	 * @return string
+	 */
+	protected function get_recurrence( $item ) {
+		$recurrence = $item->get_schedule();
+		if ( method_exists( $recurrence, 'interval_in_seconds' ) ) {
+			return self::human_interval( $recurrence->interval_in_seconds() );
+		}
+		return __( 'Non-repeating', 'action-scheduler' );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function get_items_query_limit() {
+		return $this->get_items_per_page( $this->package . '_items_per_page', $this->items_per_page );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function get_items_query_offset() {
+		global $wpdb;
+
+		$per_page = $this->get_items_query_limit();
+		$current_page = $this->get_pagenum();
+		if ( 1 < $current_page ) {
+			return $per_page * ( $current_page - 1 );
+		}
+
+		return 0;
+	}
+
+	/**
+	 * Serializes the argument of an action to render it in a human friendly format.
+	 *
+	 * @param array $row The array representation of the current row of the table
+	 *
+	 * @return string
+	 */
+	public function column_args( array $row ) {
+		return '<code>' . json_encode( $row['args'] ) . '</code>';
+	}
+
+	/**
+	 * Prints the comments, which are the log entries. It needs to be named comments otherwise it won't pickup
+	 * the WordPress styles to make the number pretty.
+	 *
+	 * It will also render all the log entries, but it will be hidden and rendered on click in the number as a modal.
+	 *
+	 * @param array $row Action array.
+	 */
+	public function column_comments( array $row ) {
+		echo '<div id="log-' . $row['ID'] . '" class="log-modal hidden" style="max-width:800px">';
+		echo '<h3>Log entries for ' . $row['ID'] . '</h3>';
```

Comment:
These strings should be passed through i18n functions so they are available for translation.

Reply:
True. I'll make it happen.